addons/mail/models/mail_mail.py

```diff
@@ -660,6 +659,9 @@ def action_send_and_close(self):
     def _send(self, auto_commit=False, raise_exception=False, smtp_session=None, alias_domain_id=False,
               mail_server=False, post_send_callback=None):
         IrMailServer = self.env['ir.mail_server']
+        if modules.module.current_test and not hasattr(IrMailServer, '_mocked_testing'):
```

Comment:
I don't like to have logic related to tests inside the code. Disabling the send in test mode is a good security, but the _mocked_testing stuff is not elegant. 

I understand that since the previous behaviour was patching `current_test`, adding more logic based on current_test will have a side effect. You don't want to commit.

A solution could be to have a something like 

```suggestion
        if self._disable_send():
```

```
def _disable_send(self):
    return modules.module.current_test
```

This makes it overridable and patchable. 
A similar change can be done in ir_mail_server.py, `if self.env['mail.mail']._disable_send()`